### ISSUE TITLE
-Werror=clobbered warning in PNG loader and saver

### ISSUE THREAD

**Issue by marxin** [godotengine/godot#26351]
As seen here:
https://travis-ci.org/godotengine/godot/jobs/499164132

```
drivers/png/image_loader_png.cpp: In function 'PoolVector<unsigned char> _lossless_pack_png(const Ref<Image>&)':
drivers/png/image_loader_png.cpp:312:6: error: variable 'cs' might be clobbered by 'longjmp' or 'vfork' [-Werror=clobbered]
  int cs = 0;
      ^~
cc1plus: all warnings being treated as errors
scons: *** [drivers/png/image_loader_png.x11.opt.tools.64.o] Error 1
drivers/png/resource_saver_png.cpp: In static member function 'static Error ResourceSaverPNG::save_image(const String&, const Ref<Image>&)':
drivers/png/resource_saver_png.cpp:104:6: error: variable 'cs' might be clobbered by 'longjmp' or 'vfork' [-Werror=clobbered]
  int cs = 0;
      ^~
cc1plus: all warnings being treated as errors
scons: *** [drivers/png/resource_saver_png.x11.opt.tools.64.o] Error 1
```

It's really problematic as these will be uninitialized once a ```longjmp``` happens:

usage:
```
	for (int i = 0; i < h; i++) {

		row_pointers[i] = (png_bytep)&r[i * w * cs];
	}
```

**Comment by marxin** [comment]
Blocks #26343

**Comment by akien-mga** [comment]
I can't seem to reproduce this error anymore with GCC 8.3.0 in the current master branch (d4178bec1f).

I'm testing re-enabling `-Wclobbered` on CI with #29803, we'll see how it goes.

**Comment by akien-mga** [comment]
It did reproduce on CI, just not in my local builds, so this is still valid.

**Comment by akien-mga** [comment]
Fixed by #29874.
